src/CreatorToken.sol

```diff
@@ -4,13 +4,22 @@ pragma solidity 0.8.20;
 import {ERC721} from "openzeppelin/token/ERC721/ERC721.sol";
 import {IERC20} from "openzeppelin/token/ERC20/IERC20.sol";
 import {SafeERC20} from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import {Pausable} from "openzeppelin/security/Pausable.sol";
 
-contract CreatorToken is ERC721 {
+contract CreatorToken is ERC721, Pausable {
   using SafeERC20 for IERC20;
 
   error CreatorToken__MaxPaymentExceeded(uint256 _price, uint256 _maxPayment);
+  error CreatorToken__CallerIsNotCreator(address _creator, address _caller);
+  error CreatorToken__CallerIsNotAdmin(address _admin, address _caller);
+  error CreatorToken__CallerIsNotCreatorOrAdmin(address _creator, address _admin, address _caller);
+  error CreatorToken__AddressZeroNotAllowed();
+  error CreatorToken__CallerIsNotOwner(uint256 _tokenId, address _owner, address _caller);
+  error CreatorToken__MinAcceptedPriceExceeded(uint256 _price, uint256 _minAcceptedPrice);
+  error CreatorToken__LastTokenCannotBeSold(uint256 _circulatingSupply);
 
   uint256 public lastId;
+  uint256 public circulatingSupply;
```

Comment:
Unless you had a specific reason for choosing this name, I think we should consider using `totalSupply` to harmonize with ERC20 instead.

Reply:
works for me, updated in https://github.com/showtime-xyz/creator-tokens/pull/13/commits/3ad5836cd07b5ec29ed6f04b6058fde8a08e27a2